OpenRA.Mods.Common/HitShapes/Rectangle.cs

```diff
@@ -67,7 +67,9 @@ public void Initialize()
 			quadrantSize = (BottomRight - TopLeft) / 2;
 			center = TopLeft + quadrantSize;
 
-			OuterRadius = new WDist(Math.Max(TopLeft.Length, BottomRight.Length));
+			var tlbr = Math.Max(TopLeft.Length, BottomRight.Length);
+			var trbl = Math.Max(new int2(BottomRight.X, TopLeft.Y).Length, new int2(TopLeft.X, BottomRight.Y).Length);
+			OuterRadius = new WDist(Math.Max(tlbr, trbl));
```

Comment:
This could be made a bit clearer, IMO:

```csharp
var topRight = new int2(BottomRight.X, TopLeft.Y);
var bottomLeft = new int2(TopLeft.X, BottomRight.Y);
var corners = new[] { TopLeft, BottomRight, topRight, bottomLeft };
OuterRadius = new WDist(corners.Select(x => x.Length).Max());
```